includes/class-wc-calypso-bridge-free-trial-payment-restrictions.php

```diff
@@ -0,0 +1,355 @@
+		add_filter( 'option_woocommerce_ppcp-oxxo-gateway_settings', function ( $value ) {
+
+			// Bail out early if the current user is allowed to create orders on free trial.
+			if ( ! is_array( $value ) || current_user_can( 'manage_woocommerce' ) ) {
+				return $value;
+			}
+
+			$value['enabled'] = 'no';
+
+			return $value;
+		}, PHP_INT_MAX );
+
+		// Disable PayPal's Pay Upon Invoice gateway for non-admin users.
+		// Couldn't test this. Given that PayPal will already be disabled from the rest of the filters, this is me being extra cautious.
+		add_filter( 'pre_option_woocommerce_ppcp-pay-upon-invoice-gateway_settings', '__return_false', PHP_INT_MAX );
+		add_filter( 'option_woocommerce_ppcp-pay-upon-invoice-gateway_settings', function ( $value ) {
+
+			// Bail out early if the current user is allowed to create orders on free trial.
+			if ( ! is_array( $value ) || current_user_can( 'manage_woocommerce' ) ) {
+				return $value;
+			}
+
+			$value['enabled'] = 'no';
+
+			return $value;
+		}, PHP_INT_MAX );
+
+	}
+
+	/**
+	 * Frontend free trial hooks.
+	 */
+	public function frontend() {
+
+		// Bail out early if the current user is allowed to create orders on free trial.
+		if ( current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		// Unset all payment gateways.
+		add_filter( 'woocommerce_available_payment_gateways', function ( $gateways ) {
+			return array();
+		}, PHP_INT_MAX );
+
+		// Change the "No available payment methods" message.
+		add_filter( 'woocommerce_no_available_payment_methods_message', function ( $message ) {
+
+			$message = __( 'This store is not ready to accept orders. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' );
+
+			return $message;
+
+		}, PHP_INT_MAX );
+
+		add_filter( 'gettext', function ( $translated_text, $text, $domain ) {
+
+			if ( $domain === 'woocommerce' ) {
+				switch ( $text ) {
+					// Since we're removing all payment gateways, we need to change the "No available payment methods" message.
+					case 'No payment method provided.' :
+						$translated_text = __( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' );
+						break;
+				}
+			}
+
+			return $translated_text;
+
+		}, PHP_INT_MAX, 3 );
+
+
+		// TODO: How do we handle translations?
+		add_action('wp_head', function(){
+			?>
+			<script type="text/javascript">
+				function myPluginGettextFilter( translation, text, domain ) {
+					if ( text === 'There are no payment methods available. This may be an error on our side. Please contact us if you need any help placing your order.' ) {
+						return 'There are no payment methods available as this store is in trial mode.';
+					}
+					return translation;
+				}
+
+				// Adding the filter
+				wp.hooks.addFilter(
+					'i18n.gettext_woocommerce',
+					'my-plugin/override-add-to-reusable-blocks-label', // TODO: What plugin name should we use?
```

Comment:
I would suggest using the `wc-calypso-bridge/override-no-payment-methods-message`